File: src/main/java/io/github/tundraclimate/finelib/addon/server/RegisterEvent.java
package io.github.tundraclimate.finelib.addon.server;

import io.github.tundraclimate.finelib.FineLib;
import org.bukkit.event.Listener;

import java.util.ArrayList;
import java.util.List;

/**
 * Implemented Register Method
 */
public abstract class RegisterEvent implements Listener {
    private static final List<RegisterEvent> register = new ArrayList<>();

    /**
     * Instance Register
     */
    public void register() {
        FineLib.getPlugin().getServer().getPluginManager().registerEvents(this, FineLib.getPlugin());
    }

    /**
     * static Register
     *
     * @param event Event
     */
    public static void register(RegisterEvent event) {
        FineLib.getPlugin().getServer().getPluginManager().registerEvents(event, FineLib.getPlugin());
    }

    public static void registerAll() {
        register.forEach(reg -> reg.register());
    }
}

File: src/main/kotlin/io/github/tundraclimate/finelib/addon/server/RegisterEvent.kt
package io.github.tundraclimate.finelib.addon.server

import io.github.tundraclimate.finelib.FineLib
import io.github.tundraclimate.finelib.addon.server.RegisterEvent
import org.bukkit.event.Listener

/**
 * Implemented Register Method
 */
interface RegisterEvent : Listener {
    /**
     * Instance Register
     */
    fun register() {
        FineLib.getPlugin().server.pluginManager.registerEvents(this, FineLib.getPlugin())
    }

    companion object {
        /**
         * static Register
         *
         * @param event Event
         */
        @JvmStatic
        fun register(event: RegisterEvent?) {
            FineLib.getPlugin().server.pluginManager.registerEvents(event!!, FineLib.getPlugin())
        }
    }
}